jsonschema-hyperschema.xml

```diff
@@ -1946,13 +1945,173 @@ Link: <https://api.example.com/trees/1/nodes/456> rev=up
+        "contextPointer": "/elements/0",
+        "rel": "self",
+        "targetUri": "https://api.example.com/things/1234",
+        "attachmentPointer": "/elements/0"
+    },
+    {
+        "contextUri": "https://api.example.com/things",
+        "contextPointer": "/elements/1",
+        "rel": "self",
+        "targetUri": "https://api.example.com/things/67890",
+        "attachmentPointer": "/elements/1"
+    }
+]]]>
+
+                    </artwork>
+                </figure>
+                <t>
+                    To fully specify our collection, we also need to add the following
+                    link to our "thing" schema.  Note that this would cause it to also
+                    appear as a link in each item in the collection representation, which
+                    is a good example of why it is best to only construct links upon request.
+                    There is no need for having as many functionally identical "collection"
+                    links as there are items in a collection page on every collection
+                    representation.
+                </t>
+                <figure>
+                    <preamble>
+                        This link would be added to the top-level "links" array in the
+                        "https://schemasexample.com/thing" schema.
+                    </preamble>
+                    <artwork>
+<![CDATA[{
+    "rel": "collection",
+    "href": "/things",
+    "targetSchema": {"$ref": "thing-collection#"},
+    "submissionSchema": {"$ref": "#"}
+}]]>
+                    </artwork>
+                    <postamble>
+                        Here we see the "submissionSchema" indicating that we can create
+                        a single "thing" by submitting a representation (minus server-created
+                        fields such as "id") to the collection that is this link's target
+                        schema.  While we cannot, in general, make assumptions about the
+                        semantics of making a data submission request (in HTTP terms, a POST),
+                        <xref target="collectionAndItem" /> tells us that we MAY make such
+                        an assumption when the link relation is "collection", and that
+                        hyper-schema authors MUST NOT use a "collection" link if the data
+                        submission operation has semantics other than item creation.
+                    </postamble>
+                </figure>
+                <t>
+                    But what if we do not have any "thing"s yet?  We cannot get to the
+                    individual "thing" schema as there is no individual "thing" to fetch.
+                    And the "tag:rel.example.com,2017:thing" link in the entry point
+                    resource does not indicate how to create a "thing".  The "self" link
+                    requires the "id" to already exist, but it is assigned by the server.
```

Comment:
Which "self" link are you talking about? I suspect this is the one in "thing" schema (which has a requirement on "id"), but it's said above that we cannot use it. On the other hand, if it's the one in "thing-collection" schema's, unless I'm missing something, this one does not a requirement on "id".

In fact, I'd use the "thing-collection" schema's "self" link as a natural place to put the `submissionSchema` to create new collection item.

Reply:
See #421 for more information about why "self" is probably inappropriate for this.

Reply:
> See #421 for more information about why "self" is probably inappropriate for this.

I see no reference to `submissionSchema` in this issue so this looks unrelated to me. This also applies the the comment below on the cref.
Maybe I missed something, would you mind elaborating a bit?

Reply:
It's actually in one of the crefs in this text that I had pasted into the issue (I'd forgotten that).

> Note also that POST-ing to a collection page that will not contain
                    the created item also seems weird.  While retrieving the collection
                    from a query parameter-less URI will still retrieve a page, that's
                    a limit imposed by the server.  POST-ing to such a URI and disregarding
                    the "default" values for the parameters seems correct.  This is another
                    reason to *not* automatically write default values into the query.

So if "self" is something like `things?offset=10&limit=10` and you POST to that to create a new "thing", if that new "thing" does not end up on that page (from the 10th to 20th "thing") then you've done something kind of strange.  You created a resource via a collection that does not actually contain the resource (there is no difference between query string parameters and path components- a change in either means a different URI and therefore a different resource).